app/src/main/java/com/itsaky/androidide/ui/compose/common/FileImage.kt

```
@@ -0,0 +1,94 @@
+	val maxDimensionPx = with(LocalDensity.current) { maxDimension.roundToPx() }
+
+	val bitmap by produceState<ImageBitmap?>(initialValue = null, file, maxDimensionPx) {
+		value =
+			file?.let { candidate ->
+				withContext(Dispatchers.IO) {
+					try {
+						if (!candidate.exists()) return@withContext null
+						decodeBounded(candidate, maxDimensionPx)?.asImageBitmap()
+					} catch (e: CancellationException) {
+						throw e
+					} catch (e: SecurityException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)